chart/compass/templates/tenant-fetcher-secret.yaml

```diff
@@ -11,5 +11,9 @@ data:
   {{ $config.secret.clientIdKey }}: {{ $config.oauth.client | b64enc | quote }}
   {{ $config.secret.clientSecretKey }}: {{ $config.oauth.secret | b64enc | quote }}
   {{ $config.secret.oauthUrlKey }}: {{ $config.oauth.tokenURL | b64enc | quote }}
+  {{ if eq $config.secret.oauthMode "oauth-mtls" }}
```

Comment:
do we need this if? oauthMode seems to be always `oauth-mtls` from values.yaml in local setup

Reply:
We actually don't.

Reply:
Update: We actually need it. If the OAuth mode is different than 'oauth-mtls' then the compass-installer will fail, because the clientCertKey and clientKeyKey do not exist.

Reply:
Do we have oauthMode different than `oauth-mtls` because in the values.yaml I see only this? And if we have different mode then I guess compass-installer fails because in the charts we have env for different modes and he try to populate all of them but in the secret we do not have the mention properties(clientCertKey, clientKeyKey)? If that's the case then we should add/remove them to/from the charts conditionally?
And also it would be semantically closer to the real environment where we have both mtls and non-mtls configuration in one secret